test/tdigest_test.rb

```diff
@@ -145,4 +154,77 @@ def test_that_it_has_a_version_number
       tdigest.size.must_equal n
     end
   end
+
+  describe '#+' do
+    it 'works with empty tdigests' do
+      other = ::TDigest::TDigest.new(0.001, 50, 1.2)
+      (tdigest + other).centroids.size.must_equal 0
+    end
+
+    describe 'adding two tdigests' do
+      before do
+        @other = ::TDigest::TDigest.new(0.001, 50, 1.2)
+        [tdigest, @other].each do |td|
+          td.push(60, 100)
+          10.times { td.push(rand * 100) }
+        end
+      end
+
+      it 'has the parameters of the left argument (the calling tdigest)' do
+        new_tdigest = tdigest + @other
+        new_tdigest.instance_variable_get(:@delta).must_equal tdigest.instance_variable_get(:@delta)
+        new_tdigest.instance_variable_get(:@k).must_equal tdigest.instance_variable_get(:@k)
+        new_tdigest.instance_variable_get(:@cx).must_equal tdigest.instance_variable_get(:@cx)
+      end
+
+      it 'results in a tdigest with number of centroids less than or equal to the combined centroids size' do
+        new_tdigest = tdigest + @other
+        new_tdigest.centroids.size.must_be :<=, tdigest.centroids.size + @other.centroids.size
+      end
+
+      it 'has the size of the two digests combined' do
+        new_tdigest = tdigest + @other
+        new_tdigest.size.must_equal (tdigest.size + @other.size)
+      end
+    end
+  end
+
+  describe '#merge!' do
+    it 'works with empty tdigests' do
+      other = ::TDigest::TDigest.new(0.001, 50, 1.2)
+      tdigest.merge!(other)
+      (tdigest).centroids.size.must_equal 0
+    end
+
+    describe 'with populated tdigests' do
+      before do
+        @other = ::TDigest::TDigest.new(0.001, 50, 1.2)
+        [tdigest, @other].each do |td|
+          td.push(60, 100)
+          10.times { td.push(rand * 100) }
+        end
+      end
+
+      it 'has the parameters of the calling tdigest' do
+        vars = [:@delta, :@k, :@cs]
+        expected = vars.map { |v| [v, tdigest.instance_variable_get(v)] }.to_h
+        tdigest.merge!(@other)
+        vars.each do |v|
+          tdigest.instance_variable_get(v).must_equal expected[v]
+        end
+      end
+
+      it 'results in a tdigest with number of centroids less than or equal to the combined centroids size' do
+        combined_size = tdigest.centroids.size + @other.centroids.size
+        tdigest.merge!(@other)
+        tdigest.centroids.size.must_be :<=, combined_size
+      end
+
+      it 'has the size of the two digests combined' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.